lib/waterwheel.js

```diff
@@ -1,37 +1,51 @@
 const Request = require('./helpers/request');
 const Entity = require('./entity');
-
 const JSONAPI = require('./jsonapi');
-
 const methods = require('./helpers/methods');
 
 module.exports = class Waterwheel extends Request {
   /**
    * Create an instance of the Waterwheel class.
-   * @param {string} base
-   *   The base path of the Drupal site.
-   * @param {object} credentials
-   *   The credentials object.
-   * @param {object} resources
-   *   An object representing all the resources available to waterwheel.
+   * @param {object} options
+   *   The configuration used to create a new instance of Waterwheel.
+   * @param {string} options.base
+   *   The base URL.
+   * @param {object} options.credentials
+   *   The credentials used with each request.
+   * @param {string} options.credentials.oauth
+   *   The OAuth2 Bearer token.
+   * @param {object} options.resources
+   *   A JSON object representing all the resources available to Waterwheel.
+   * @param {string} options.timeout
+   *   How long AXIOS should wait before bailing on a request.
    */
-  constructor(base, credentials, resources = null) {
-    if (base === '' || !base) {
+  constructor(options) {
+    if (!options.hasOwnProperty('base')) {
       throw new Error('Missing base path.');
     }
 
-    super(base, credentials);
+    if (!options.hasOwnProperty('credentials')) {
+      throw new Error('Missing credentials.');
+    }
 
-    this.base = base;
-    this.api = {};
+    super(options);
 
-    this.jsonapi = new JSONAPI(this.base, credentials);
+    this.api = {};
+    this.jsonapi = new JSONAPI(options);
 
-    if (resources && Object.keys(resources).length) {
-      Object.keys(resources).forEach(entity => {
+    if (this.options.resources && Object.keys(this.options.resources).length) {
+      Object.keys(this.options.resources).forEach(entity => {
         this.api[entity] = {};
-        (resources[entity].bundles ? resources[entity].bundles : [resources[entity].label.toLowerCase().replace(/ /g, '_')]).forEach(bundle => {
-          this.api[entity][bundle] = new Entity(this.base, this.credentials, resources[entity].methods, entity, bundle, resources[entity].more);
+        (this.options.resources[entity].bundles ? this.options.resources[entity].bundles : [this.options.resources[entity].label.toLowerCase().replace(/ /g, '_')]).forEach(bundle => {
+          this.api[entity][bundle] = new Entity({
+            base: this.options.base,
+            credentials: this.options.credentials,
+            methods: this.options.resources[entity].methods,
+            entity: entity,
+            bundle: bundle,
+            more: this.options.resources[entity].more,
```

Comment:
`more` is very unclear to me. Is there a way to rename this so it's a bit more descriptive?